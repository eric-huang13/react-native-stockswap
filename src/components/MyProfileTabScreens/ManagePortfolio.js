import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  TextInput,
  SafeAreaView,
  FlatList,
} from 'react-native';
import SearchInput from '../../icons/SearchInput';
import TriangleIcon from '../../icons/TriangleIcon';
import {connect} from 'react-redux';
import ManagePortfolioBox from '../MyProfileTabComponents/ManagePortfolioBox';
import {moderateScale} from '../../util/responsiveFont';
import {PortfolioAccounts} from '../../actions/profile';
import InstitutionHoldingsCard from './InstitutionHoldingsCard';


class ManagePortfolio extends Component {
  componentDidMount() {
    this.props.PortfolioAccounts();
  }
  constructor(props) {
    super(props);
    this.state = {
      input: '',
      shouldShow: false,
      dropDown: 'Select sorting',
    };
  }
  handleChange = (text) => {
    this.setState({input: text});
  };

  dropDownSelect(pick) {
    this.setState({dropDown: pick, shouldShow: false});
  }
  render() {
    const {gainers} = this.props;

    const filteredStocks = gainers.filter(
      (item) =>
        item.title.toLowerCase().includes(this.state.input.toLowerCase()) ||
        item.symbol.toLowerCase().includes(this.state.input.toLowerCase()),
    );

    const {shouldShow} = this.state;
    if (!this.props.portfolioAccounts.institutions) {
      return (
        <SafeAreaView style={style.container}>        
        </SafeAreaView>
      );    }
    return (
      <SafeAreaView style={style.container}>
        <View style={style.searchInputContainer}>
          <View style={style.searchInputInnerContainer}>
            <SearchInput />
          </View>
          <TextInput
            style={style.searchInput}
            placeholder="Search"
            placeholderTextColor="lightgrey"
            onChangeText={(text) => this.handleChange(text)}
          />
        </View>
        <View style={style.innerContainer}>
          <View style={style.topDetailsRow}>
            <View style={style.percentContainer}>
              <Text style={style.portfolio}>Portfolio</Text>
              <Text style={style.percent}>$3,201</Text>
            </View>
            <View style={style.gainDetailsContainer}>
              <Text style={style.gain}>$-10.75(-11%)</Text>
              <Text style={style.gainTime}>Past day</Text>
            </View>
          </View>
          <View style={style.percentButtonContainer}>
            <Text style={style.stockHeader}>STOCKS</Text>
            <View style={style.dotsDropdownContainer}>
              <TouchableOpacity
                onPress={() =>
                  this.setState({
                    shouldShow: !shouldShow,
                  })
                }>
                <View style={style.percentChangeContainer}>
                  <Text style={style.percentChangeButton}>
                    {this.state.dropDown}
                  </Text>
                  <TriangleIcon style={style.icon} />
                </View>
              </TouchableOpacity>
              {this.state.shouldShow ? (
                <View style={style.dropdown}>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect('Percent Change')}>
                    <Text style={style.dropDownText}>Percent Change</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect('Last price')}>
                    <Text style={style.dropDownText}>Last price</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect('Total percent change')}>
                    <Text style={style.dropDownText}>Total percent change</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect('Your equity')}>
                    <Text style={style.dropDownText}>Your equity</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect("Today's return")}>
                    <Text style={style.dropDownText}>Today's return</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect('Total return')}>
                    <Text style={style.dropDownText}>Total return</Text>
                  </TouchableOpacity>
                </View>
              ) : null}
            </View>
          </View>
          {/* <FlatList
            style={style.boxContainer}
            data={filteredStocks}
            renderItem={({item}) => (
              <TouchableOpacity
                key={item.id}
                onPress={() =>
                  this.props.navigation.navigate({
                    name: 'ManagePortfolioCompany',
                    params: {item},
                  })
                }>
                <View key={item.id} style={style.portfolioBoxContainer}>
                  <ManagePortfolioBox item={item} />
                </View>
              </TouchableOpacity>
            )}
          /> */}
          <View>
            <ScrollView style={style.scrollContainer}>
            {this.props.portfolioAccounts.institutions.map((item, index) => (
              <InstitutionHoldingsCard
                key={index}
                itemId={item.itemId}
                insId={item.institutionId}
              />
 
            ))}
            </ScrollView>
          </View>
        </View>
      </SafeAreaView>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    gainers: state.company.gainers,
    portfolioAccounts: state.user.portfolioAccounts,

  };
};
const mapDispatchToProps = (dispatch) => {
  return {
    PortfolioAccounts: () => dispatch(PortfolioAccounts()),
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(ManagePortfolio);

const style = StyleSheet.create({
  container: {
    backgroundColor: '#2a334a',
    paddingBottom:315,
    flex:1

  },
  scrollContainer:{
// flex:1
  },
  searchInputContainer: {
    marginBottom: moderateScale(22),
  },
  searchInput: {
    paddingLeft: moderateScale(40),
    alignContent: 'center',
    backgroundColor: '#3e4d6c',
    color: 'lightgrey',
    fontSize: moderateScale(15),
    height: moderateScale(36),
    fontFamily: 'Montserrat-Italic',
    paddingVertical: 0,
  },
  searchInputInnerContainer: {
    position: 'absolute',
    zIndex: 1,
    left: moderateScale(14),
    top: moderateScale(10),
  },
  innerContainer: {
    paddingHorizontal: moderateScale(6),
  },
  topDetailsRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  gainDetailsContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: moderateScale(4),
  },
  gain: {
    fontFamily: 'Montserrat-Medium',
    color: '#F66E6E',
    fontSize: moderateScale(12),
  },
  gainTime: {
    fontFamily: 'Montserrat-Medium',
    color: 'lightgrey',
    fontSize: moderateScale(10),
    marginLeft: moderateScale(8),
  },
  boxContainer: {
    marginTop: moderateScale(6),  

  },
  portfolioBoxContainer: {
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(158, 150, 150, .6)',
    padding: moderateScale(0.1),
    marginTop: moderateScale(8),
  },
  percentContainer: {
    paddingHorizontal: moderateScale(8),
  },
  portfolio: {
    color: '#FFFFFF',
    fontSize: moderateScale(16),
    fontFamily: 'Montserrat-Regular',
  },
  percent: {
    color: '#FFFFFF',
    fontSize: moderateScale(24),
    fontFamily: 'Montserrat-ExtraBold',
  },
  percentButtonContainer: {
    marginTop: moderateScale(12),
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: moderateScale(8),
  },
  stockHeader: {
    color: '#FFFFFF',
    fontSize: moderateScale(22),
    fontFamily: 'Montserrat-Bold',
  },
  percentChangeButton: {
    color: '#FFFFFF',
    fontSize: moderateScale(14),
    backgroundColor: '#3E4D6C',
    borderRadius: moderateScale(6),
    fontFamily: 'Montserrat-Italic',
    marginLeft: moderateScale(6),
    width: moderateScale(190),
  },
  percentChangeContainer: {
    borderRadius: moderateScale(6),
    backgroundColor: '#3E4D6C',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: moderateScale(30),
  },

  dropdown: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: moderateScale(29),
    backgroundColor: '#3E4D6C',
    zIndex: 1,
    paddingVertical: moderateScale(6),
    height: moderateScale(200),
    position: 'absolute',
  },
  dropDownText: {
    color: 'white',
    fontSize: moderateScale(16),
    marginHorizontal: moderateScale(12),
    fontFamily: 'Montserrat-Medium',
    marginBottom: moderateScale(6),
  },
  dropDownTextReportContainer: {
    borderTopWidth: moderateScale(1),
    borderTopColor: 'lightgrey',
    paddingTop: moderateScale(4),
    backgroundColor: '#2C3957',
  },
  icon: {
    marginRight: moderateScale(4),
  },
});
