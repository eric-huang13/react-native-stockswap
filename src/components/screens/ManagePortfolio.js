import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  TextInput,
  SafeAreaView,
  FlatList,
} from 'react-native';
import SearchInput from '../../icons/SearchInput';
import TriangleIcon from '../../icons/TriangleIcon';
import {connect} from 'react-redux';
import ManagePortfolioBox from './ManagePortfolioBox';

class ManagePortfolio extends Component {
  constructor(props) {
    super(props);
    this.state = {
      input: '',
      shouldShow: false,
      dropDown: 'Select sorting',
    };
  }
  handleChange = (text) => {
    this.setState({input: text});
  };

  dropDownSelect(pick) {
    this.setState({dropDown: pick, shouldShow: false});
  }
  render() {
    const {gainers} = this.props;

    const filteredStocks = gainers.filter(
      (item) =>
        item.title.toLowerCase().includes(this.state.input.toLowerCase()) ||
        item.symbol.toLowerCase().includes(this.state.input.toLowerCase()),
    );

    const {shouldShow} = this.state;

    return (
      <SafeAreaView style={style.container}>
        <View style={style.searchInputContainer}>
          <View style={style.searchInputInnerContainer}>
            <SearchInput />
          </View>
          <TextInput
            style={style.searchInput}
            placeholder="Search"
            placeholderTextColor="lightgrey"
            onChangeText={(text) => this.handleChange(text)}
          />
        </View>
        <View style={style.innerContainer}>
          <View style={style.topDetailsRow}>
            <View style={style.percentContainer}>
              <Text style={style.portfolio}>Portfolio</Text>
              <Text style={style.percent}>$3,201</Text>
            </View>
            <View style={style.gainDetailsContainer}>
              <Text style={style.gain}>$-10.75(-11%)</Text>
              <Text style={style.gainTime}>Past hour</Text>
            </View>
          </View>
          <View style={style.percentButtonContainer}>
            <Text style={style.stockHeader}>STOCKS</Text>
            <View style={style.dotsDropdownContainer}>
              <TouchableOpacity
                onPress={() =>
                  this.setState({
                    shouldShow: !shouldShow,
                  })
                }>
                <View style={style.percentChangeContainer}>
                  <Text style={style.percentChangeButton}>
                    {this.state.dropDown}
                  </Text>
                  <TriangleIcon style={style.icon} />
                </View>
              </TouchableOpacity>
              {this.state.shouldShow ? (
                <View style={style.dropdown}>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect('Percent Change')}>
                    <Text style={style.dropDownText}>Percent Change</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect('Last price')}>
                    <Text style={style.dropDownText}>Last price</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect('Total percent change')}>
                    <Text style={style.dropDownText}>Total percent change</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect('Your equity')}>
                    <Text style={style.dropDownText}>Your equity</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect("Today's return")}>
                    <Text style={style.dropDownText}>Today's return</Text>
                  </TouchableOpacity>
                  <TouchableOpacity
                    onPress={() => this.dropDownSelect('Total return')}>
                    <Text style={style.dropDownText}>Total return</Text>
                  </TouchableOpacity>
                </View>
              ) : null}
            </View>
          </View>
          <FlatList
            style={style.boxContainer}
            data={filteredStocks}
            renderItem={({item}) => (
              <TouchableOpacity
                key={item.id}
                onPress={() =>
                  this.props.navigation.navigate({
                    name: 'ManagePortfolioCompany',
                    params: {item},
                  })
                }>
                <View key={item.id} style={style.portfolioBoxContainer}>
                  <ManagePortfolioBox item={item} />
                </View>
              </TouchableOpacity>
            )}
          />
        </View>
      </SafeAreaView>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    gainers: state.company.gainers,
  };
};
export default connect(mapStateToProps)(ManagePortfolio);

const style = StyleSheet.create({
  container: {
    backgroundColor: '#2a334a',
    flex: 1,
  },
  searchInputContainer: {
    marginBottom: 22,
  },
  searchInput: {
    paddingLeft: 40,
    alignContent: 'center',
    backgroundColor: '#3e4d6c',
    color: 'lightgrey',
    fontSize: 15,
    height: 36,
    fontFamily: 'Montserrat-Italic',
    paddingVertical: 0,
  },
  searchInputInnerContainer: {
    position: 'absolute',
    zIndex: 1,
    left: 14,
    top: 10,
  },
  innerContainer: {
    paddingHorizontal: 6,
  },
  topDetailsRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  gainDetailsContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 4,
  },
  gain: {
    fontFamily: 'Montserrat-Medium',
    color: '#F66E6E',
    fontSize: 12,
  },
  gainTime: {
    fontFamily: 'Montserrat-Medium',
    color: 'lightgrey',
    fontSize: 10,
    marginLeft: 8,
  },
  boxContainer: {
    marginTop: 6,
  },
  portfolioBoxContainer: {
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(158, 150, 150, .6)',
    padding: 0.1,
    marginTop: 8,
  },
  percentContainer: {
    paddingHorizontal: 8,
  },
  portfolio: {
    color: '#FFFFFF',
    fontSize: 16,
    fontFamily: 'Montserrat-Regular',
  },
  percent: {
    color: '#FFFFFF',
    fontSize: 24,
    fontFamily: 'Montserrat-ExtraBold',
  },
  percentButtonContainer: {
    marginTop: 12,
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 8,
  },
  stockHeader: {
    color: '#FFFFFF',
    fontSize: 22,
    fontFamily: 'Montserrat-Bold',
  },
  percentChangeButton: {
    color: '#FFFFFF',
    fontSize: 14,
    backgroundColor: '#3E4D6C',
    borderRadius: 6,
    fontFamily: 'Montserrat-Italic',
    marginLeft: 6,
    width: 190,
  },
  percentChangeContainer: {
    borderRadius: 6,
    backgroundColor: '#3E4D6C',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: 30,
  },

  dropdown: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: 29,
    backgroundColor: '#3E4D6C',
    zIndex: 1,
    paddingVertical: 6,
    height: 200,
    position: 'absolute',
  },
  dropDownText: {
    color: 'white',
    fontSize: 16,
    marginHorizontal: 12,
    fontFamily: 'Montserrat-Medium',
    marginBottom: 6,
  },
  dropDownTextReportContainer: {
    borderTopWidth: 1,
    borderTopColor: 'lightgrey',
    paddingTop: 4,
    backgroundColor: '#2C3957',
  },
  icon: {
    marginRight: 4,
  },
});
