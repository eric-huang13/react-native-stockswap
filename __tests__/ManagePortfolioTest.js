import React from 'react';
import ManagePortfolio from '../src/components/MyProfileTabScreens/ManagePortfolio';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render   (<ManagePortfolio gainers={[]} />);
    const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});