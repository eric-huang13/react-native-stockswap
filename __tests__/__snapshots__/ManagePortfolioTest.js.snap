// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 31.072992700729927,
      }
    }
  >
    <View
      style={
        Object {
          "left": 19.77372262773723,
          "position": "absolute",
          "top": 14.124087591240876,
          "zIndex": 1,
        }
      }
    >
      <SearchInput />
    </View>
    <Component
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search"
      placeholderTextColor="lightgrey"
      rejectResponderTermination={true}
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#3e4d6c",
          "color": "lightgrey",
          "fontFamily": "Montserrat-Italic",
          "fontSize": 21.186131386861312,
          "height": 50.846715328467155,
          "paddingLeft": 56.496350364963504,
          "paddingVertical": 0,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 8.474452554744525,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 11.299270072992702,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 22.598540145985403,
            }
          }
        >
          Portfolio
        </Text>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-ExtraBold",
              "fontSize": 33.8978102189781,
            }
          }
        >
          $3,201
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "paddingHorizontal": 5.649635036496351,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#F66E6E",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 16.94890510948905,
            }
          }
        >
          $-10.75(-11%)
        </Text>
        <Text
          style={
            Object {
              "color": "lightgrey",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 14.124087591240876,
              "marginLeft": 11.299270072992702,
            }
          }
        >
          Past hour
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 16.94890510948905,
          "paddingHorizontal": 11.299270072992702,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 31.072992700729927,
          }
        }
      >
        STOCKS
      </Text>
      <View>
        <ForwardRef
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3E4D6C",
                "borderRadius": 8.474452554744525,
                "flexDirection": "row",
                "height": 42.372262773722625,
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "#3E4D6C",
                  "borderRadius": 8.474452554744525,
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-Italic",
                  "fontSize": 19.77372262773723,
                  "marginLeft": 8.474452554744525,
                  "width": 268.35766423357666,
                }
              }
            >
              Select sorting
            </Text>
            <TriangleIcon
              style={
                Object {
                  "marginRight": 5.649635036496351,
                }
              }
            />
          </View>
        </ForwardRef>
      </View>
    </View>
    <FlatList
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      style={
        Object {
          "marginTop": 8.474452554744525,
        }
      }
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
</ForwardRef(SafeAreaView)>
`;
